### PR TITLE
android/ui: add support for remembering the last used exit node

### PR DESCRIPTION
**Pull request by barnstar**
Updates ENG-2911

Disabling an exit node is now temporary and you can re-enable it without re-selecting it from the picker.

Selecting "none" will clear the current selection as well as the previously remembered node.

### CODE REVIEW

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt [hunk 1]
```diff
@@ -60,17 +60,30 @@ class MainViewModel : IpnViewModel() {
...
-    Client(viewModelScope).editPrefs(prefsOut) { LoadingIndicator.stop() }
+    if (prefs.activeExitNodeID != null) {
+      // We have an active exit node so we should keep it, but disable it
+      Client(viewModelScope).setUseExitNode(false) { LoadingIndicator.stop() }
+    } else if (prefs.selectedExitNodeID != null) {
+      // We have a prior exit node to enable
+      Client(viewModelScope).setUseExitNode(true) { LoadingIndicator.stop() }
+    } else {
+      // This should not be possible.  In this state the button is hidden
+      Log.e(TAG, "No exit node to disable an no prior exit node to enable")
```
Comment (agottardo) [inline android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt:85]:
```suggestion
      Log.e(TAG, "No exit node to disable and no prior exit node to enable")
```

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt
Comment (agottardo) [inline android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt:1]: We're starting to add a lot of exit node-specific logic in here, perhaps we can move it out to a separate component one day?

File: android/src/main/java/com/tailscale/ipn/ui/theme/Theme.kt [hunk 2]
```diff
@@ -333,6 +333,17 @@ val ColorScheme.onBackgroundLogoDotDisabled: Color
         Color(0x66FFFFFF)
       }
 
+val ColorScheme.button: ButtonColors
```
Comment (agottardo) [inline android/src/main/java/com/tailscale/ipn/ui/theme/Theme.kt:336]: Should we call this `exitNodeBannerButton`?

**Review by agottardo — APPROVED** [review]
Some nits

**Review by kari-ts — APPROVED** [review]

**Pull request merged**